src/act.informative.cpp

```diff
@@ -3563,8 +3563,8 @@ ACMD(do_spells) {
         }
     }
 
-    /* Is this character in a class with spells? */
-    if (MEM_MODE(tch) == MEM_NONE) {
+    /* Is this character in a class or a race with spells? */
+    if (!IS_SPELLCASTER_CLASS(tch) && !IS_SPELLCASTER_RACE(tch)) {
```

Comment:
This is an and, not an or.  I think you want || to have it return true if their class OR race is magical

Reply:
No, this is an "and".  They should only be told they can't cast spells if they are both not a spellcasting class AND not a spellcasting race.  Otherwise if you're a non-spellcasting race but still a spellcasting class, you won't see any spells via the spells command!